library/management/permission/src/main/java/org/quiltmc/qsl/permission/api/Permissions.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2020 lucko (Luck) <luck@lucko.me>
+ * Copyright 2022 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.permission.api;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.util.Identifier;
+import org.jetbrains.annotations.NotNull;
+import org.quiltmc.qsl.base.api.util.TriState;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+public class Permissions {
+
+	/**
+	 * Gets the {@link TriState state} of a {@code permission} for the given source.
+	 *
+	 * @param source the source to perform the check for
+	 * @param permission the permission to check
+	 * @return the state of the permission
+	 */
+	@NotNull
+	public static TriState getPermissionValue(@NotNull ServerCommandSource source, @NotNull Identifier permission) {
```

Comment:
i feel like there's a better name than this, but i don't know what it would be

Reply:
You could probably shorten it to `getPermission` or even just `get` (especially if `TriState` was renamed to `PermissionState` as I suggested above).